src/mat4.c

```diff
@@ -0,0 +1,73 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   mat4.c                                             :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/11/29 15:00:10 by gwolf             #+#    #+#             */
+/*   Updated: 2023/12/20 11:35:44 by gwolf            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "mat4.h"
+#include "ft_print.h"
+
+t_mat4	ft_mat4_set_identity(void)
+{
+	static const t_mat4	identity = {
```

Comment:
The C99 standard guarantees it:
6.7.8 paragraph 19 "[...] all subobjects that are not initialized explicitly shall be initialized implicitly the same as objects that have static storage duration."
And 6.7.8. paragraph 10: "[...] If an object that has static storage duration is not initialized explicitly,
then: [...] if it has arithmetic type, it is initialized to (positive or unsigned) zero;"

Reply:
So whenever an arithmetic variable is defined static, it is initialised to zero?

Reply:
Yes 
- if you have `static int count`, it is init to 0
- if you have `static bool switch` it is init to false (since false is 0)

But: since they are static any subsequent function call will have the previous value.

Reply:
Good